pkg/storage/cloud/external_storage.go

```diff
@@ -518,39 +556,74 @@ func checkHTTPContentRangeHeader(h string, pos int64) error {
 	return nil
 }
 
+func (r *resumingHTTPReader) sendRequest(
+	reqHeaders map[string]string,
+) (resp *http.Response, err error) {
+	prev := timeutil.Now()
+	attempt := 0
+	for retries := retry.StartWithCtx(r.ctx, httpRetryOptions); retries.Next(); {
+		attempt++
+		resp, err = r.client.req(r.ctx, "GET", r.url, nil, reqHeaders)
+		if resp != nil {
+			r.body = resp.Body
+		}
+
+		delay := timeutil.Since(prev)
+		prev = timeutil.Now()
+
+		if err == nil {
```

Comment:
can more this above the time calls, right? (or just remove the timing altogether -- the log lines themselves have timestamps)